### PR TITLE
Pin controller-gen to v0.7.0

### PR DESCRIPTION
**Pull request by ncdc**
- Now installing controller-gen in a consistent location for everyone hacking on this project - thanks to the Cluster API folks for the install script!
- Pinned controller-gen to v0.7.0
- Remove code-generator from hack/tools.go so it's not listed in our project's go.mod any more
- hack/update-codegen.sh now must be invoked via `make` (it errors if not)
- Fixed codegen script to work with controller-gen v0.7.0
- Regenerated CRDs using updated controller-gen version

### CODE REVIEW

**Review by stevekuznetsov — APPROVED** [review]
LGTM, will merge on green

**Comment by stevekuznetsov** [comment]
Actually - @ncdc can you add a verify script and run it in CI, please?

**Comment by ncdc** [comment]
You want verify-codegen?

**Comment by stevekuznetsov** [comment]
I want my PR to fail to merge unless I re-gen with the correct version :)

**Comment by ncdc** [comment]
k

**Comment by ncdc** [comment]
Updated, PTAL

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -4,22 +4,17 @@ set -o errexit
...
+if [[ -z "${MAKELEVEL:-}" ]]; then
+    echo 'You must invoke this script via make'
+    exit 1
 fi
 
-"$( dirname "${BASH_SOURCE[0]}")/update-codegen-clients.sh"
+"$(dirname "${BASH_SOURCE[0]}")/update-codegen-clients.sh"
 
 # Update generated CRD YAML
-${GOPATH}/bin/controller-gen crd:preserveUnknownFields=false rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/
```
Comment (davidfestal) [inline hack/update-codegen.sh:25]: What's the reason for removing `:preserveUnknownFields=false` ?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -57,20 +57,21 @@ jobs:
...
       with:
         go-version: v1.16
     - name: Download modules
       run: go mod download
-    - name: Check generated clients
-      run: hack/verify-codegen-clients.sh
+    - name: Check codegen
+      run: make verify-codegen
     - run: make build
+    # Have to check kubernetes out after verify-codegen
```
Comment (davidfestal) [inline .github/workflows/ci.yaml:68]: Could we add the why, in the comment ?

File: hack/update-codegen.sh (hunk 1 above)
Comment (stevekuznetsov) [reply hack/update-codegen.sh:25]: I think the option is not exposed in future versions of the tool

**Review by stevekuznetsov — APPROVED** [review]
This LGTM!

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (ncdc) [reply .github/workflows/ci.yaml]: Updated, PTAL

**Review by davidfestal — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (davidfestal) [reply .github/workflows/ci.yaml]: Thanks !

**Pull request merged**